### ISSUE TITLE
EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'

### ISSUE THREAD

**Issue by longzheng** [seydx/homebridge-camera-ui#362]
**Describe the bug**
After running `homebridge-camera-ui` with 7 cameras with `videoanalysis` and `hksv`, I seem to be experiencing an issue where it has errors clearing timers and the plugin appears to stop working.

It appears the error is related to having too many file handlers open to the `database.json` file.

When I encounter the issue, I'm also unable to open the camera.ui web app as it errors loading everything

![image](https://user-images.githubusercontent.com/484912/149757681-991f3bc4-4170-4af3-8b9a-71c54dc2ff22.png)

**To Reproduce**
- Run homebridge for several hours
- Encounter issue in logs

**Expected behavior**

**Logs**
```
[17/01/2022, 9:43:10 pm] [CameraUI] Intercom: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'
[17/01/2022, 9:43:10 pm] [CameraUI] Intercom: Camera "Intercom" not found.
[17/01/2022, 9:43:11 pm] [CameraUI] Cleartimer: An error occured during removing notification (bdd9e2f2e1) due to cleartimer
[17/01/2022, 9:43:11 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[17/01/2022, 9:43:37 pm] [CameraUI] VIDEOANALYSIS: New message: Data: {"time":"2022-01-17T10:43:37.118Z","event":"dwellTime"} - Motion: resetted - Camera: Living
[17/01/2022, 9:43:59 pm] [CameraUI] Living: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'
[17/01/2022, 9:43:59 pm] [CameraUI] Living: Camera "Living" not found.
[17/01/2022, 9:45:01 pm] [CameraUI] Cleartimer: An error occured during removing notification (fcc246a036) due to cleartimer
[17/01/2022, 9:45:01 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[17/01/2022, 9:46:24 pm] [CameraUI] Cleartimer: An error occured during removing notification (8edb68cf3e) due to cleartimer
[17/01/2022, 9:46:24 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[17/01/2022, 9:48:20 pm] [CameraUI] Cleartimer: An error occured during removing notification (50b9ce6593) due to cleartimer
[17/01/2022, 9:48:20 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[17/01/2022, 9:51:42 pm] [CameraUI] Cleartimer: An error occured during removing notification (f0e7cb5dee) due to cleartimer
[17/01/2022, 9:51:42 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}

```

**Screenshots**

**Environment**
- Node.js Version: v16.13.2
- Homebridge Version: v1.4.0-beta.7
- Camera UI Version: v5.0.9
- Operating System: Windows 11
- Browser: Chrome

**Comment by seydx** [comment]
Hey @longzheng 

Can you please try `v5.0.10-beta.0`, I have removed unnecessary actions that manipulate the database, that might fix your problem too. And it seems that the problem occurs only on Windows? I have run several systems with dummy cameras for a "stress test" and it works fine for several days. I have not tested it on Windows though. But it would be great if we could fix the problem for Windows as well.

**Comment by longzheng** [comment]
Thanks. Testing now.

**Comment by longzheng** [comment]
I'm seeing much better results, not erroring as consistently, but I still noticed a couple instances for example
```
[18/01/2022, 12:30:00 pm] [CameraUI] Living: Motion OFF - Motion handler timeout.
[18/01/2022, 12:30:01 pm] [CameraUI] Side: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:01 pm] [CameraUI] Side: Motion data: [{"zone":"region0","percent":11.900887489318848,"sensitivity":89,"dwell":60}]}
[18/01/2022, 12:30:03 pm] [CameraUI] Side: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:03 pm] [CameraUI] Side: Motion data: [{"zone":"region0","percent":11.818567276000977,"sensitivity":89,"dwell":60}]}
[18/01/2022, 12:30:06 pm] [CameraUI] Backyard: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:06 pm] [CameraUI] Backyard: Motion data: [{"zone":"region0","percent":57.10980987548828,"sensitivity":44,"dwell":60}]}
[18/01/2022, 12:30:07 pm] [CameraUI] Backyard: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:07 pm] [CameraUI] Backyard: Motion data: [{"zone":"region0","percent":21.35850715637207,"sensitivity":80,"dwell":60}]}
[18/01/2022, 12:30:07 pm] [CameraUI] Backyard: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:07 pm] [CameraUI] Backyard: Motion data: [{"zone":"region0","percent":12.383246421813965,"sensitivity":89,"dwell":60}]}
[18/01/2022, 12:30:08 pm] [CameraUI] Front: An error occured during handling motion event
[18/01/2022, 12:30:08 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[18/01/2022, 12:30:08 pm] [CameraUI] Living: Recording completed (HSV)
[18/01/2022, 12:30:08 pm] [CameraUI] Living: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:08 pm] [CameraUI] Living: Motion data: [{"zone":"region0","percent":10.338541984558105,"sensitivity":91,"dwell":60}]}
[18/01/2022, 12:30:08 pm] [CameraUI] Backyard: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:08 pm] [CameraUI] Backyard: Motion data: [{"zone":"region0","percent":12.453125,"sensitivity":89,"dwell":60}]}
[18/01/2022, 12:30:09 pm] [CameraUI] Living: Closing recording process
[18/01/2022, 12:30:09 pm] [CameraUI] Living: FFmpeg recording process exited (expected)
[18/01/2022, 12:30:20 pm] [CameraUI] Laundry: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:20 pm] [CameraUI] Laundry: Motion data: [{"zone":"region0","percent":10.112726211547852,"sensitivity":91,"dwell":60}]}
[18/01/2022, 12:30:32 pm] [CameraUI] Laundry: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:32 pm] [CameraUI] Laundry: Motion data: [{"zone":"region0","percent":11.186897277832031,"sensitivity":90,"dwell":60}]}
[18/01/2022, 12:30:40 pm] [CameraUI] Backyard: An error occured during handling motion event
[18/01/2022, 12:30:40 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[18/01/2022, 12:30:41 pm] [CameraUI] Side: An error occured during handling motion event
[18/01/2022, 12:30:41 pm] [CameraUI] [Error: EMFILE: too many open files, open 'C:\Users\Long\.homebridge\camera.ui\database\database.json'] {
  errno: -4066,
  code: 'EMFILE',
  syscall: 'open',
  path: 'C:\\Users\\Long\\.homebridge\\camera.ui\\database\\database.json'
}
[18/01/2022, 12:30:47 pm] [CameraUI] Side: New motion detected, resetting motion in 60s..
[18/01/2022, 12:30:47 pm] [CameraUI] Side: Motion data: [{"zone":"region0","percent":11.297420501708984,"sensitivity":90,"dwell":60}]}
```

**Comment by seydx** [comment]
@longzheng 

v5.0.10-beta.3 is online

**Comment by seydx** [comment]
should be fixed in v5.0.10-beta.5

**Comment by seydx** [comment]
v5.0.10 is online
